ape_vyper/flatten.py

```
@@ -0,0 +1,46 @@
+from typing import Optional
+
+
+def flatten(path: str, base_path: Optional[str] = None) -> str:
+    """
+    Function for flattening a Vyper file from a given path to a string.
+
+    Args:
+        path (str): Path to the Vyper file
+        base_path (Optional[str]): Base path to the Vyper file
+
+    Returns:
+        str: the flattened Vyper file
+    """
+    flattened = ""
+
+    with open(path, "r") as file:
+        for line in file.readlines():
+            if "import" in line:
+                if base_path is None:
+                    raise ValueError(
+                        "Base path must be provided when flattening a file with imports."
+                    )
+                # Parse the import path from the Vyper File
+                import_path = base_path + _get_import_path(line)
+                flattened += flatten(import_path, base_path)
```

If there are SPDX license IDs in the interfaces files AND in the main file, we are going to want to remove them. There can only 1 license ID in the final flattened file. Also, if you encounter differing licenses, you can raise an exception as that is a bit problematic.